[🐛 BUG] DataNode not found error appearing for no reason

### What went wrong? 🤔

This error appears for no reason. The Data Node and CSV have been created in the `.taipy` folder. I believe there are no errors in my code. I am using *default_data*, so the issue might be here.

```
[2024-06-17 18:11:56.972][Taipy][ERROR] DataNode not found: DATANODE_input_1_data_node_e5934b26-895c-43b5-ba5d-e3c2b23e5f5f
[2024-06-17 18:11:56.974][Taipy][ERROR] DataNode not found: DATANODE_input_1_data_node_e5934b26-895c-43b5-ba5d-e3c2b23e5f5f
[2024-06-17 18:11:56.978][Taipy][ERROR] DataNode not found: DATANODE_input_1_data_node_e5934b26-895c-43b5-ba5d-e3c2b23e5f5f
```

### Expected Behavior

No error should be reported. These errors are false positives.

### Steps to Reproduce Issue

```python
# Import necessary libraries
from taipy import Config
import taipy as tp
import json


# Function to run a Dataiku scenario
def run_something(input_1):
   return input_1

with open('data/model_variables_default.json') as f:
    model_variables_default = json.load(f)

input_1_cfg = Config.configure_csv_data_node(id="input_1_data_node", default_data=model_variables_default)

output_cfg = Config.configure_data_node(id="output_data_node")



# Scenario and task configuration in Taipy
scenario_task_cfg = Config.configure_task(
    id="scenario_task",
    function=run_something,
    input=[input_1_cfg],
    output=[output_cfg])


scenario_cfg = Config.configure_scenario(
    id="scenario",
    task_configs=[scenario_task_cfg])


# Main execution block with GUI setup
if __name__ == "__main__":
    tp.Core().run()
    scenario = tp.create_scenario(scenario_cfg)
    # Run the GUI
    import time
    time.sleep(10)
```
### Version of Taipy

develop - 6/17/24


### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.


### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

--- a/taipy/core/data/csv.py
+++ b/taipy/core/data/csv.py
@@ -18,6 +18,7 @@ import pandas as pd
 
 from taipy.config.common.scope import Scope
 
+from .._entity._reload import _Reloader
 from .._version._version_manager_factory import _VersionManagerFactory
 from ..job.job_id import JobId
 from ._file_datanode_mixin import _FileDataNodeMixin
@@ -116,7 +117,8 @@ class CSVDataNode(DataNode, _FileDataNodeMixin, _TabularDataNodeMixin):
             **properties,
         )
 
-        self._write_default_data(default_value)
+        with _Reloader():
+            self._write_default_data(default_value)
 
         self._TAIPY_PROPERTIES.update(
             {

--- a/taipy/core/data/excel.py
+++ b/taipy/core/data/excel.py
@@ -18,6 +18,7 @@ from openpyxl import load_workbook
 
 from taipy.config.common.scope import Scope
 
+from .._entity._reload import _Reloader
 from .._version._version_manager_factory import _VersionManagerFactory
 from ..exceptions.exceptions import ExposedTypeLengthMismatch, NonExistingExcelSheet, SheetNameLengthMismatch
 from ..job.job_id import JobId
@@ -118,7 +119,8 @@ class ExcelDataNode(DataNode, _FileDataNodeMixin, _TabularDataNodeMixin):
             **properties,
         )
 
-        self._write_default_data(default_value)
+        with _Reloader():
+            self._write_default_data(default_value)
 
         self._TAIPY_PROPERTIES.update(
             {

--- a/taipy/core/data/json.py
+++ b/taipy/core/data/json.py
@@ -18,7 +18,7 @@ from typing import Any, Dict, List, Optional, Set
 
 from taipy.config.common.scope import Scope
 
-from .._entity._reload import _self_reload
+from .._entity._reload import _Reloader, _self_reload
 from .._version._version_manager_factory import _VersionManagerFactory
 from ._file_datanode_mixin import _FileDataNodeMixin
 from .data_node import DataNode
@@ -112,7 +112,8 @@ class JSONDataNode(DataNode, _FileDataNodeMixin):
         self._decoder = self._properties.get(self._DECODER_KEY, _DefaultJSONDecoder)
         self._encoder = self._properties.get(self._ENCODER_KEY, _DefaultJSONEncoder)
 
-        self._write_default_data(default_value)
+        with _Reloader():
+            self._write_default_data(default_value)
 
         self._TAIPY_PROPERTIES.update(
             {

--- a/taipy/core/data/parquet.py
+++ b/taipy/core/data/parquet.py
@@ -18,6 +18,7 @@ import pandas as pd
 
 from taipy.config.common.scope import Scope
 
+from .._entity._reload import _Reloader
 from .._version._version_manager_factory import _VersionManagerFactory
 from ..exceptions.exceptions import UnknownCompressionAlgorithm, UnknownParquetEngine
 from ..job.job_id import JobId
@@ -153,7 +154,8 @@ class ParquetDataNode(DataNode, _FileDataNodeMixin, _TabularDataNodeMixin):
             **properties,
         )
 
-        self._write_default_data(default_value)
+        with _Reloader():
+            self._write_default_data(default_value)
 
         if not self._last_edit_date and (isfile(self._path) or isdir(self._path)):
             self._last_edit_date = datetime.now()

--- a/taipy/core/data/pickle.py
+++ b/taipy/core/data/pickle.py
@@ -15,6 +15,7 @@ from typing import List, Optional, Set
 
 from taipy.config.common.scope import Scope
 
+from .._entity._reload import _Reloader
 from .._version._version_manager_factory import _VersionManagerFactory
 from ._file_datanode_mixin import _FileDataNodeMixin
 from .data_node import DataNode
@@ -98,7 +99,8 @@ class PickleDataNode(DataNode, _FileDataNodeMixin):
             **properties,
         )
 
-        self._write_default_data(default_value)
+        with _Reloader():
+            self._write_default_data(default_value)
 
         self._TAIPY_PROPERTIES.update(
             {

